packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1GatewayRouter.sol

```diff
@@ -229,6 +229,14 @@ contract L1GatewayRouter is WhitelistConsumer, L1ArbitrumMessenger, GatewayRoute
         uint256 _gasPriceBid,
         bytes calldata _data
     ) public payable override onlyWhitelisted returns (bytes memory) {
+        // when sending a L1 to L2 transaction, we expect the user to send
+        // eth in flight in order to pay for L2 gas costs
+        // this check prevents users from misconfiguring the msg.value
+        require(msg.value > 0, "NO_ETH_FOR_GAS");
```

Comment:
Thoughts on leaving this check in or maybe replacing it with a `_maxSubmissionCost > 0`